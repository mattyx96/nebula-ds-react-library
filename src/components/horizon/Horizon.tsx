import { useEffect, useRef, useState } from 'react';

type Props = {
  color?: string;
  numLines?: number;
  lineThickness?: number;
  distance?: number;
  distanceGrowthFactor?: number;
  thicknessDecayFactor?: number;
  className?: string;
  height?: number;
  width?: number;
  inverse?: boolean;
};

export const Horizon = ({
                          color = '#000',
                          numLines = 55,
                          lineThickness = 10,
                          distance = 10,
                          distanceGrowthFactor = 1.05,
                          thicknessDecayFactor = 0.95,
                          height,
                          width,
                          ...props
                        }: Props) => {
  const svgRef = useRef<SVGSVGElement>(null);
  const [svgWidth, setSvgWidth] = useState(1512); // default width

  const generatePaths = () => {
    const paths = [];
    let yOffset, currentThickness;

    if (props.inverse) {
      yOffset = (distance + distanceGrowthFactor) * (numLines - 1);
      currentThickness = lineThickness;
    } else {
      yOffset = 0;
      currentThickness = lineThickness;
    }

    for (let i = 0; i < numLines; i++) {
      paths.push(
        <path
          key={i}
          d={`M0 ${yOffset}H${svgWidth}`}
          stroke={color}
          strokeWidth={currentThickness}
        />
      );

      if (props.inverse) {
        yOffset -= distance + distanceGrowthFactor; // Decrease the distance with growth factor
      } else {
        yOffset += distance + distanceGrowthFactor; // Increase the distance with growth factor
      }
      currentThickness *= thicknessDecayFactor; // Decrease the thickness
    }

    return paths;
  };

  const fitSvgPathElements = () => {
    if (svgRef.current) {
      const pathElements = svgRef.current.querySelectorAll('path');
      if (pathElements.length > 0) {
        let minX = Infinity;
        let minY = Infinity;
        let maxX = -Infinity;
        let maxY = -Infinity;

        pathElements.forEach((pathElement) => {
          const pathBox = pathElement.getBBox();
          minX = Math.min(minX, pathBox.x);
          minY = Math.min(minY, pathBox.y);
          maxX = Math.max(maxX, pathBox.x + pathBox.width);
          maxY = Math.max(maxY, pathBox.y + pathBox.height);
        });

        const svgHeight = height || (maxY - minY);
        const currentSvgWidth = width || svgRef.current.clientWidth || (maxX - minX);

        if (svgRef.current) {
          svgRef.current.setAttribute('height', `${svgHeight}`);
          svgRef.current.setAttribute('width', `${currentSvgWidth}`);
        }

        setSvgWidth(currentSvgWidth); // Update state with the current SVG width
      }
    }
  };

  useEffect(() => {
    window.addEventListener('resize', fitSvgPathElements);

    return () => {
      window.removeEventListener('resize', fitSvgPathElements);
    };
  }, []);

  useEffect(() => {
    fitSvgPathElements();
  }, [color, numLines, lineThickness, distance, distanceGrowthFactor, thicknessDecayFactor]);

  useEffect(() => {
    fitSvgPathElements(); // Initial call on load
  }, []);

  return (
    <svg
      ref={svgRef}
      className={props.className}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      {generatePaths()}
    </svg>
  );
};
