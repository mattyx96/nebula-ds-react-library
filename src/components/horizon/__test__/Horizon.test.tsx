import {render} from '@testing-library/react';
import {describe, expect, it, beforeAll, vi} from 'vitest';
import {Horizon} from '../Horizon';

// Extend the SVGElement type to include the getBBox method
declare global {
  interface SVGElement {
    getBBox(): { x: number; y: number; width: number; height: number };
  }
}

describe('Horizon', () => {
  // Mock getBBox method
  beforeAll(() => {
    SVGElement.prototype.getBBox = vi.fn().mockImplementation(() => ({
      x: 0,
      y: 0,
      width: 100,
      height: 100,
    }));
  });

  it('renders correctly with default props', () => {
    const {container} = render(<Horizon/>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with className prop', () => {
    const {container} = render(<Horizon className="custom-class"/>);
    expect(container.firstChild).toHaveClass('custom-class');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const {container} = render(<Horizon color="red"/>);
    const paths = container.querySelectorAll('path');
    paths.forEach(path => {
      expect(path).toHaveAttribute('stroke', 'red');
    });
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom number of lines', () => {
    const {container} = render(<Horizon numLines={30}/>);
    const paths = container.querySelectorAll('path');
    expect(paths).toHaveLength(30);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom line thickness', () => {
    const lineThickness = 20;
    const thicknessDecayFactor = 0.95;
    const numLines = 30;

    const {container} = render(
      <Horizon lineThickness={lineThickness} thicknessDecayFactor={thicknessDecayFactor} numLines={numLines}/>
    );

    const paths = container.querySelectorAll('path');
    let currentThickness = lineThickness;
    paths.forEach((path) => {
      expect(path).toHaveAttribute('stroke-width', currentThickness.toString());
      currentThickness *= thicknessDecayFactor; // Update the thickness for the next path
    });
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom distance and distance growth factor', () => {
    const {container} = render(<Horizon distance={15} distanceGrowthFactor={1.1}/>);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom thickness decay factor', () => {
    const {container} = render(<Horizon thicknessDecayFactor={0.9}/>);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with inverse prop', () => {
    const {container} = render(<Horizon inverse/>);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom height and width', () => {
    const {container} = render(<Horizon height={500} width={500}/>);
    const svgElement = container.querySelector('svg');
    expect(svgElement).toHaveAttribute('height', '500');
    expect(svgElement).toHaveAttribute('width', '500');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with all props mixed', () => {
    const props = {
      color: 'blue',
      numLines: 40,
      lineThickness: 15,
      distance: 20,
      distanceGrowthFactor: 1.2,
      thicknessDecayFactor: 0.85,
      className: 'custom-class',
      height: 600,
      width: 800,
      inverse: true,
    };
    const {container} = render(<Horizon {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
