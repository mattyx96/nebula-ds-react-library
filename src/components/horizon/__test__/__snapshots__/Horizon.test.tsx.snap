// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Horizon > renders correctly with all props mixed 1`] = `
<svg
  class="custom-class"
  fill="none"
  height="600"
  width="800"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 826.8H800"
    stroke="blue"
    stroke-width="15"
  />
  <path
    d="M0 805.5999999999999H800"
    stroke="blue"
    stroke-width="12.75"
  />
  <path
    d="M0 784.3999999999999H800"
    stroke="blue"
    stroke-width="10.8375"
  />
  <path
    d="M0 763.1999999999998H800"
    stroke="blue"
    stroke-width="9.211875000000001"
  />
  <path
    d="M0 741.9999999999998H800"
    stroke="blue"
    stroke-width="7.8300937500000005"
  />
  <path
    d="M0 720.7999999999997H800"
    stroke="blue"
    stroke-width="6.6555796875"
  />
  <path
    d="M0 699.5999999999997H800"
    stroke="blue"
    stroke-width="5.657242734375"
  />
  <path
    d="M0 678.3999999999996H800"
    stroke="blue"
    stroke-width="4.80865632421875"
  />
  <path
    d="M0 657.1999999999996H800"
    stroke="blue"
    stroke-width="4.087357875585937"
  />
  <path
    d="M0 635.9999999999995H800"
    stroke="blue"
    stroke-width="3.4742541942480463"
  />
  <path
    d="M0 614.7999999999995H800"
    stroke="blue"
    stroke-width="2.9531160651108395"
  />
  <path
    d="M0 593.5999999999995H800"
    stroke="blue"
    stroke-width="2.5101486553442136"
  />
  <path
    d="M0 572.3999999999994H800"
    stroke="blue"
    stroke-width="2.1336263570425813"
  />
  <path
    d="M0 551.1999999999994H800"
    stroke="blue"
    stroke-width="1.813582403486194"
  />
  <path
    d="M0 529.9999999999993H800"
    stroke="blue"
    stroke-width="1.5415450429632649"
  />
  <path
    d="M0 508.79999999999933H800"
    stroke="blue"
    stroke-width="1.3103132865187752"
  />
  <path
    d="M0 487.59999999999934H800"
    stroke="blue"
    stroke-width="1.113766293540959"
  />
  <path
    d="M0 466.39999999999935H800"
    stroke="blue"
    stroke-width="0.946701349509815"
  />
  <path
    d="M0 445.19999999999936H800"
    stroke="blue"
    stroke-width="0.8046961470833427"
  />
  <path
    d="M0 423.9999999999994H800"
    stroke="blue"
    stroke-width="0.6839917250208413"
  />
  <path
    d="M0 402.7999999999994H800"
    stroke="blue"
    stroke-width="0.5813929662677151"
  />
  <path
    d="M0 381.5999999999994H800"
    stroke="blue"
    stroke-width="0.4941840213275578"
  />
  <path
    d="M0 360.3999999999994H800"
    stroke="blue"
    stroke-width="0.4200564181284241"
  />
  <path
    d="M0 339.1999999999994H800"
    stroke="blue"
    stroke-width="0.35704795540916046"
  />
  <path
    d="M0 317.99999999999943H800"
    stroke="blue"
    stroke-width="0.30349076209778636"
  />
  <path
    d="M0 296.79999999999944H800"
    stroke="blue"
    stroke-width="0.2579671477831184"
  />
  <path
    d="M0 275.59999999999945H800"
    stroke="blue"
    stroke-width="0.21927207561565065"
  />
  <path
    d="M0 254.39999999999947H800"
    stroke="blue"
    stroke-width="0.18638126427330304"
  />
  <path
    d="M0 233.19999999999948H800"
    stroke="blue"
    stroke-width="0.15842407463230757"
  />
  <path
    d="M0 211.9999999999995H800"
    stroke="blue"
    stroke-width="0.13466046343746144"
  />
  <path
    d="M0 190.7999999999995H800"
    stroke="blue"
    stroke-width="0.11446139392184222"
  />
  <path
    d="M0 169.5999999999995H800"
    stroke="blue"
    stroke-width="0.09729218483356589"
  />
  <path
    d="M0 148.39999999999952H800"
    stroke="blue"
    stroke-width="0.082698357108531"
  />
  <path
    d="M0 127.19999999999952H800"
    stroke="blue"
    stroke-width="0.07029360354225135"
  />
  <path
    d="M0 105.99999999999952H800"
    stroke="blue"
    stroke-width="0.059749563010913644"
  />
  <path
    d="M0 84.79999999999951H800"
    stroke="blue"
    stroke-width="0.0507871285592766"
  />
  <path
    d="M0 63.59999999999951H800"
    stroke="blue"
    stroke-width="0.043169059275385106"
  />
  <path
    d="M0 42.39999999999951H800"
    stroke="blue"
    stroke-width="0.03669370038407734"
  />
  <path
    d="M0 21.19999999999951H800"
    stroke="blue"
    stroke-width="0.03118964532646574"
  />
  <path
    d="M0 -4.902744876744691e-13H800"
    stroke="blue"
    stroke-width="0.026511198527495877"
  />
</svg>
`;

exports[`Horizon > renders correctly with className prop 1`] = `
<div>
  <svg
    class="custom-class"
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 16.2H100"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.4H100"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.599999999999994H100"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.8H100"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 81H100"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 97.2H100"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.4H100"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 129.6H100"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 145.79999999999998H100"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 161.99999999999997H100"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 178.19999999999996H100"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 194.39999999999995H100"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 210.59999999999994H100"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 226.79999999999993H100"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom color 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="red"
      stroke-width="13"
    />
    <path
      d="M0 16.2H100"
      stroke="red"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.4H100"
      stroke="red"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.599999999999994H100"
      stroke="red"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.8H100"
      stroke="red"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 81H100"
      stroke="red"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 97.2H100"
      stroke="red"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.4H100"
      stroke="red"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 129.6H100"
      stroke="red"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 145.79999999999998H100"
      stroke="red"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 161.99999999999997H100"
      stroke="red"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 178.19999999999996H100"
      stroke="red"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 194.39999999999995H100"
      stroke="red"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 210.59999999999994H100"
      stroke="red"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 226.79999999999993H100"
      stroke="red"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom distance and distance growth factor 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 16.1H100"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.2H100"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.300000000000004H100"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.4H100"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 80.5H100"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 96.6H100"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 112.69999999999999H100"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 128.79999999999998H100"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 144.89999999999998H100"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 160.99999999999997H100"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 177.09999999999997H100"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 193.19999999999996H100"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 209.29999999999995H100"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 225.39999999999995H100"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom height and width 1`] = `
<div>
  <svg
    fill="none"
    height="500"
    width="500"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H500"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 16.2H500"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.4H500"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.599999999999994H500"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.8H500"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 81H500"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 97.2H500"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.4H500"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 129.6H500"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 145.79999999999998H500"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 161.99999999999997H500"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 178.19999999999996H500"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 194.39999999999995H500"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 210.59999999999994H500"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 226.79999999999993H500"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom line thickness 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="#000"
      stroke-width="20"
    />
    <path
      d="M0 16.2H100"
      stroke="#000"
      stroke-width="19"
    />
    <path
      d="M0 32.4H100"
      stroke="#000"
      stroke-width="18.05"
    />
    <path
      d="M0 48.599999999999994H100"
      stroke="#000"
      stroke-width="17.1475"
    />
    <path
      d="M0 64.8H100"
      stroke="#000"
      stroke-width="16.290125"
    />
    <path
      d="M0 81H100"
      stroke="#000"
      stroke-width="15.475618749999999"
    />
    <path
      d="M0 97.2H100"
      stroke="#000"
      stroke-width="14.701837812499999"
    />
    <path
      d="M0 113.4H100"
      stroke="#000"
      stroke-width="13.966745921874999"
    />
    <path
      d="M0 129.6H100"
      stroke="#000"
      stroke-width="13.268408625781248"
    />
    <path
      d="M0 145.79999999999998H100"
      stroke="#000"
      stroke-width="12.604988194492185"
    />
    <path
      d="M0 161.99999999999997H100"
      stroke="#000"
      stroke-width="11.974738784767576"
    />
    <path
      d="M0 178.19999999999996H100"
      stroke="#000"
      stroke-width="11.376001845529196"
    />
    <path
      d="M0 194.39999999999995H100"
      stroke="#000"
      stroke-width="10.807201753252736"
    />
    <path
      d="M0 210.59999999999994H100"
      stroke="#000"
      stroke-width="10.266841665590098"
    />
    <path
      d="M0 226.79999999999993H100"
      stroke="#000"
      stroke-width="9.753499582310592"
    />
    <path
      d="M0 242.99999999999991H100"
      stroke="#000"
      stroke-width="9.265824603195062"
    />
    <path
      d="M0 259.19999999999993H100"
      stroke="#000"
      stroke-width="8.802533373035308"
    />
    <path
      d="M0 275.3999999999999H100"
      stroke="#000"
      stroke-width="8.362406704383542"
    />
    <path
      d="M0 291.5999999999999H100"
      stroke="#000"
      stroke-width="7.944286369164364"
    />
    <path
      d="M0 307.7999999999999H100"
      stroke="#000"
      stroke-width="7.547072050706145"
    />
    <path
      d="M0 323.9999999999999H100"
      stroke="#000"
      stroke-width="7.169718448170838"
    />
    <path
      d="M0 340.1999999999999H100"
      stroke="#000"
      stroke-width="6.811232525762295"
    />
    <path
      d="M0 356.39999999999986H100"
      stroke="#000"
      stroke-width="6.470670899474181"
    />
    <path
      d="M0 372.59999999999985H100"
      stroke="#000"
      stroke-width="6.1471373545004715"
    />
    <path
      d="M0 388.79999999999984H100"
      stroke="#000"
      stroke-width="5.839780486775448"
    />
    <path
      d="M0 404.99999999999983H100"
      stroke="#000"
      stroke-width="5.547791462436675"
    />
    <path
      d="M0 421.1999999999998H100"
      stroke="#000"
      stroke-width="5.270401889314841"
    />
    <path
      d="M0 437.3999999999998H100"
      stroke="#000"
      stroke-width="5.0068817948490985"
    />
    <path
      d="M0 453.5999999999998H100"
      stroke="#000"
      stroke-width="4.756537705106643"
    />
    <path
      d="M0 469.7999999999998H100"
      stroke="#000"
      stroke-width="4.518710819851311"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom number of lines 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 16.2H100"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.4H100"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.599999999999994H100"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.8H100"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 81H100"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 97.2H100"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.4H100"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 129.6H100"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 145.79999999999998H100"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 161.99999999999997H100"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 178.19999999999996H100"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 194.39999999999995H100"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 210.59999999999994H100"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 226.79999999999993H100"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
    <path
      d="M0 242.99999999999991H100"
      stroke="#000"
      stroke-width="2.6765847172304382"
    />
    <path
      d="M0 259.19999999999993H100"
      stroke="#000"
      stroke-width="2.4089262455073945"
    />
    <path
      d="M0 275.3999999999999H100"
      stroke="#000"
      stroke-width="2.1680336209566553"
    />
    <path
      d="M0 291.5999999999999H100"
      stroke="#000"
      stroke-width="1.9512302588609898"
    />
    <path
      d="M0 307.7999999999999H100"
      stroke="#000"
      stroke-width="1.7561072329748908"
    />
    <path
      d="M0 323.9999999999999H100"
      stroke="#000"
      stroke-width="1.5804965096774017"
    />
    <path
      d="M0 340.1999999999999H100"
      stroke="#000"
      stroke-width="1.4224468587096617"
    />
    <path
      d="M0 356.39999999999986H100"
      stroke="#000"
      stroke-width="1.2802021728386956"
    />
    <path
      d="M0 372.59999999999985H100"
      stroke="#000"
      stroke-width="1.152181955554826"
    />
    <path
      d="M0 388.79999999999984H100"
      stroke="#000"
      stroke-width="1.0369637599993435"
    />
    <path
      d="M0 404.99999999999983H100"
      stroke="#000"
      stroke-width="0.9332673839994092"
    />
    <path
      d="M0 421.1999999999998H100"
      stroke="#000"
      stroke-width="0.8399406455994683"
    />
    <path
      d="M0 437.3999999999998H100"
      stroke="#000"
      stroke-width="0.7559465810395215"
    />
    <path
      d="M0 453.5999999999998H100"
      stroke="#000"
      stroke-width="0.6803519229355693"
    />
    <path
      d="M0 469.7999999999998H100"
      stroke="#000"
      stroke-width="0.6123167306420124"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with custom thickness decay factor 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0H100"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 16.2H100"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 32.4H100"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 48.599999999999994H100"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 64.8H100"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 81H100"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 97.2H100"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.4H100"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 129.6H100"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 145.79999999999998H100"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 161.99999999999997H100"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 178.19999999999996H100"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 194.39999999999995H100"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 210.59999999999994H100"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 226.79999999999993H100"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;

exports[`Horizon > renders correctly with default props 1`] = `
<svg
  fill="none"
  height="100"
  width="100"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0H100"
    stroke="#000"
    stroke-width="13"
  />
  <path
    d="M0 16.2H100"
    stroke="#000"
    stroke-width="11.700000000000001"
  />
  <path
    d="M0 32.4H100"
    stroke="#000"
    stroke-width="10.530000000000001"
  />
  <path
    d="M0 48.599999999999994H100"
    stroke="#000"
    stroke-width="9.477000000000002"
  />
  <path
    d="M0 64.8H100"
    stroke="#000"
    stroke-width="8.529300000000003"
  />
  <path
    d="M0 81H100"
    stroke="#000"
    stroke-width="7.676370000000003"
  />
  <path
    d="M0 97.2H100"
    stroke="#000"
    stroke-width="6.9087330000000025"
  />
  <path
    d="M0 113.4H100"
    stroke="#000"
    stroke-width="6.217859700000003"
  />
  <path
    d="M0 129.6H100"
    stroke="#000"
    stroke-width="5.596073730000002"
  />
  <path
    d="M0 145.79999999999998H100"
    stroke="#000"
    stroke-width="5.036466357000002"
  />
  <path
    d="M0 161.99999999999997H100"
    stroke="#000"
    stroke-width="4.532819721300002"
  />
  <path
    d="M0 178.19999999999996H100"
    stroke="#000"
    stroke-width="4.079537749170002"
  />
  <path
    d="M0 194.39999999999995H100"
    stroke="#000"
    stroke-width="3.671583974253002"
  />
  <path
    d="M0 210.59999999999994H100"
    stroke="#000"
    stroke-width="3.3044255768277018"
  />
  <path
    d="M0 226.79999999999993H100"
    stroke="#000"
    stroke-width="2.9739830191449315"
  />
</svg>
`;

exports[`Horizon > renders correctly with inverse prop 1`] = `
<div>
  <svg
    fill="none"
    height="100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 226.79999999999998H100"
      stroke="#000"
      stroke-width="13"
    />
    <path
      d="M0 210.6H100"
      stroke="#000"
      stroke-width="11.700000000000001"
    />
    <path
      d="M0 194.4H100"
      stroke="#000"
      stroke-width="10.530000000000001"
    />
    <path
      d="M0 178.20000000000002H100"
      stroke="#000"
      stroke-width="9.477000000000002"
    />
    <path
      d="M0 162.00000000000003H100"
      stroke="#000"
      stroke-width="8.529300000000003"
    />
    <path
      d="M0 145.80000000000004H100"
      stroke="#000"
      stroke-width="7.676370000000003"
    />
    <path
      d="M0 129.60000000000005H100"
      stroke="#000"
      stroke-width="6.9087330000000025"
    />
    <path
      d="M0 113.40000000000005H100"
      stroke="#000"
      stroke-width="6.217859700000003"
    />
    <path
      d="M0 97.20000000000005H100"
      stroke="#000"
      stroke-width="5.596073730000002"
    />
    <path
      d="M0 81.00000000000004H100"
      stroke="#000"
      stroke-width="5.036466357000002"
    />
    <path
      d="M0 64.80000000000004H100"
      stroke="#000"
      stroke-width="4.532819721300002"
    />
    <path
      d="M0 48.60000000000004H100"
      stroke="#000"
      stroke-width="4.079537749170002"
    />
    <path
      d="M0 32.400000000000034H100"
      stroke="#000"
      stroke-width="3.671583974253002"
    />
    <path
      d="M0 16.200000000000035H100"
      stroke="#000"
      stroke-width="3.3044255768277018"
    />
    <path
      d="M0 3.552713678800501e-14H100"
      stroke="#000"
      stroke-width="2.9739830191449315"
    />
  </svg>
</div>
`;
